integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JmsToBackendWithInboundEndpointTestCase.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.esb.jms.inbound.transport.test;
+
+import org.apache.axiom.om.OMElement;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.annotations.ExecutionEnvironment;
+import org.wso2.carbon.automation.engine.annotations.SetEnvironment;
+import org.wso2.carbon.automation.extensions.servers.jmsserver.client.JMSQueueMessageProducer;
+import org.wso2.carbon.automation.extensions.servers.jmsserver.controller.config.JMSBrokerConfigurationProvider;
+import org.wso2.carbon.integration.common.admin.client.LogViewerClient;
+import org.wso2.carbon.logging.view.stub.types.carbon.LogEvent;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.servers.ActiveMQServer;
+
+import java.io.File;
+
+/**
+ * This testcase tests consuming message from a queue and sending to a backend works
+ */
+public class JmsToBackendWithInboundEndpointTestCase extends ESBIntegrationTest {
+
+    private static final String QUEUE_NAME = "jmsQueueToHttpWithInboundEndpointTestCase";
+    private ActiveMQServer activeMQServer = new ActiveMQServer();
+
+    @BeforeClass(alwaysRun = true)
+    public void initialize() throws Exception {
+        activeMQServer.startJMSBroker();
+        super.init();
+
+        OMElement esbSynapseConfig = esbUtils.loadResource(
+                File.separator + "artifacts" + File.separator + "ESB" + File.separator + "jms" + File.separator +
+                        "inbound" + File.separator + "transport" + File.separator + "jmsQueueToHttpInboundSynapse.xml");
+        updateESBConfiguration(esbSynapseConfig);
```

Comment:
Is this aligning with new test framework?

Reply:
Thanks. I will do the necessary changes.